Source/WebCore/layout/formattingContexts/grid/ImplicitGrid.cpp

```diff
@@ -100,24 +100,25 @@ void ImplicitGrid::insertUnplacedGridItem(const UnplacedGridItem& unplacedGridIt
 
 }
 
-PlacedGridItems ImplicitGrid::placedGridItems() const
+ImplicitGrid::GridAreas ImplicitGrid::gridAreas() const
 {
-    HashSet<UnplacedGridItem> processedUnplacedGridItems;
-    PlacedGridItems placedGridItems;
+    GridAreas gridAreas;
+    gridAreas.reserveInitialCapacity(rowsCount() * columnsCount());
 
     for (size_t rowIndex = 0; rowIndex < m_gridMatrix.size(); ++rowIndex) {
         for (size_t columnIndex = 0; columnIndex < m_gridMatrix[rowIndex].size(); ++columnIndex) {
 
             const auto& gridCell = m_gridMatrix[rowIndex][columnIndex];
             for (const auto& unplacedGridItem : gridCell) {
-                if (processedUnplacedGridItems.contains(unplacedGridItem))
+                if (gridAreas.contains(unplacedGridItem))
                     continue;
```

Comment:
I think you are still double-hashing. The `HashSet::ensure` step will only enter the lambda if the key argument passed is not in the container. So you shouldn't need this early `continue` step, as the call to `gridAreas.ensure(...)` will just immediately bail out after checking if the item is already in the container.

Reply:
Oops you are right! Embarrassing!